src/main/java/com/microsoft/sqlserver/jdbc/ISQLServerEnclaveProvider.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.sqlserver.jdbc;
+
+import java.security.InvalidAlgorithmParameterException;
+import java.security.NoSuchAlgorithmException;
+import java.security.PrivateKey;
+import java.util.ArrayList;
+
+
+public interface ISQLServerEnclaveProvider {
+    byte[] getEnclavePackage(String userSQL, ArrayList<byte[]> enclaveCEKs) throws SQLServerException;
+
+    void getAttestationParamters(boolean b,
+            String s) throws NoSuchAlgorithmException, InvalidAlgorithmParameterException;
+
+    void createEnclaveSession(SQLServerConnection connection, String userSql, String preparedTypeDefinitions,
+            Parameter[] params, ArrayList<String> parameterNames) throws SQLServerException;
+
+    void invalidateEnclaveSession();
+
+    EnclaveSession getEnclaveSession();
+}
+
+
+abstract class BaseAttestationRequest {
+    protected PrivateKey privateKey;
+
+    byte[] getBytes() {
+        return null;
+    };
+}
+
+
+class EnclaveSession {
+    private byte[] sessionID;
+    private long counter;
+    private byte[] sessionSecret;
+
+    // Don't allow default instantiation
+    @SuppressWarnings("unused")
+    private EnclaveSession() {};
+
+    public EnclaveSession(byte[] cs, byte[] b) {
+        sessionID = cs;
+        sessionSecret = b;
+        counter = 0;
+    }
+
+    byte[] getSessionID() {
+        return sessionID;
+    }
+
+    byte[] getSessionSecret() {
+        return sessionSecret;
+    }
+
+    long getCounter() {
+        counter++;

```

Comment:
Please add a comment that explains this.

Reply:
perhaps a better name? typically getters don't change values